scripts/libraries/ml/ml/preprocessing.py

```diff
@@ -42,12 +44,32 @@ def __call__(self, *args):
             raise ValueError("Expected batches to be 1")
         if c != 1 and c != 3:
             raise ValueError("Expected channels to be 1 or 3")
+
+        # Place the image buffer at the end of the input buffer so we can convert it in-place.
         pixfmt = image.GRAYSCALE if c == 1 else image.RGB565
-        img = image.Image(w, h, pixfmt, buffer=buffer)
+        offset = len(buffer) - (w * h * (1 if c == 1 else 2))
+        img = image.Image(w, h, pixfmt, buffer=memoryview(buffer)[offset:])
```

Comment:
I was able to change the loops in img.ndarray() to walk forward by moving the image buffer to the end of the tensor area. This allows an in-place conversion. This works because image pixels are either the same size or smaller than ndarray "pixels".

Reply:
Is this for the cache prefetch optimization we were discussing ?

Reply:
Yeah, this should fix that issue. I was listening.